examples/sensors/imu.py

```diff
@@ -0,0 +1,139 @@
+from custom_data_handlers import IS_MATPLOTLIB_AVAILABLE, IS_PYQTGRAPH_AVAILABLE, MPLPlotter, PyQtGraphPlotter
+from tqdm import tqdm
+
+import genesis as gs
+from genesis.sensors.data_handlers import NPZFileWriter
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-v", "--vis", action="store_true", default=True)
+    parser.add_argument("-c", "--cpu", action="store_true", default=False)
+    args = parser.parse_args()
+
+    ########################## init ##########################
+    gs.init(backend=gs.cpu if args.cpu else gs.gpu)
+
+    ########################## create a scene ##########################
+    scene = gs.Scene(
+        viewer_options=gs.options.ViewerOptions(
+            camera_pos=(3.5, 0.0, 2.5),
+            camera_lookat=(0.0, 0.0, 0.5),
+            camera_fov=40,
+        ),
+        sim_options=gs.options.SimOptions(
+            dt=0.01,
+        ),
+        profiling_options=gs.options.ProfilingOptions(
+            show_FPS=False,
+        ),
+        show_viewer=args.vis,
+    )
+
+    ########################## entities ##########################
+    scene.add_entity(gs.morphs.Plane())
+    franka = scene.add_entity(
+        gs.morphs.MJCF(file="xml/franka_emika_panda/panda.xml"),
+    )
+    end_effector = franka.get_link("hand")
+    motors_dof = np.arange(7)
+
+    imu = scene.add_sensor(
+        gs.sensors.IMU(
+            entity_idx=franka.idx,
+            link_idx_local=end_effector.idx_local,
+            # noise parameters
+            acc_axes_skew=(0.0, 0.01, 0.02),
+            gyro_axes_skew=(0.03, 0.04, 0.05),
+            acc_noise_std=(0.01, 0.01, 0.01),
+            gyro_noise_std=(0.01, 0.01, 0.01),
+            acc_random_walk_std=(0.001, 0.001, 0.001),
+            gyro_random_walk_std=(0.001, 0.001, 0.001),
+            delay=0.01,
+            jitter=0.01,
+            interpolate=True,
+        )
+    )
+    labels = {"lin_acc": ["acc_x", "acc_y", "acc_z"], "ang_vel": ["gyro_x", "gyro_y", "gyro_z"]}
+    if IS_PYQTGRAPH_AVAILABLE:
+        imu.add_recording(
+            gs.options.RecordingOptions(handler=PyQtGraphPlotter(title="IMU Measured Data", labels=labels))
+        )
+        imu.add_recording(
+            gs.options.RecordingOptions(handler=PyQtGraphPlotter(title="IMU Ground Truth Data", labels=labels)),
+            read_ground_truth=True,
+        )
+    elif IS_MATPLOTLIB_AVAILABLE:
+        imu.add_recording(
+            gs.options.RecordingOptions(
+                handler=MPLPlotter(title="IMU Measured Data", labels=labels),
+                run_in_thread=False,
+            )
+        )
+        imu.add_recording(
+            gs.options.RecordingOptions(
+                handler=MPLPlotter(title="IMU Ground Truth Data", labels=labels),
+                run_in_thread=False,
+            ),
+            read_ground_truth=True,
+        )
+    else:
+        print("matplotlib or pyqtgraph not found, skipping real-time plotting.")
```

Comment:
Why is it so much more complex in this case than in the other example? Could `run_in_thread` be deduced automatically from the DataHandler itself if not specific? By adding a base read-only property `def support_threading(self):` (or a simple class attribute if you prefer but it is less flexible as it may be option-dependent). This way, threading would be enabled by default if supported, disabled otherwise, and you can raise a clear exception in case of incompatible options. The other advantage is that running in thread or not is no longer a concern for the average Joe.

This is basically what we do for the interactive viewer :)